RequiredExports.h

```diff
@@ -77,41 +77,3 @@ struct BufferDesc
 		bufferDesc.bufferStart = 0; \
 		bufferDesc.length = 0; \
 	}
-
-
-// ** Deprecated **
-// ****************
-#ifndef NoDeprecated
-
-#define SCRIPT_API extern "C" __declspec(dllexport)

```

Comment:
My two multiplayer scenarios use the SCRIPT_API to make the AI player work. I suspect removing this line will affect other scenarios as well.

```C++
SCRIPT_API SDescBlockEx DescBlockEx = { 1, 0, 0, 0, 0, 0, 0, 0 };
```

Is the solution to just write this instead?

```C++
extern "C" __declspec(dllexport) SDescBlockEx DescBlockEx = { 1, 0, 0, 0, 0, 0, 0, 0 };
```

I don't see any reason to keep this macro in service if this is all it is used for. Except that it is referenced a lot in tutorials and posts in the forums. If we want to remove it, how to add AI should probably be documented properly somewhere (Wiki maybe?).

@DanRStevens, thoughts?

-Brett

Reply:
When that section was deprecated, a renamed macro was added to replace it:
https://github.com/OutpostUniverse/Outpost2DLL/blob/72391d9cbe9c056a8d704ec2c2837ce36f75c6bb/RequiredExports.h#L8

I think the newer name is more clear about what it does.

Partially updated code would then be:
```cpp
Export SDescBlockEx DescBlockEx = { 1, 0, 0, 0, 0, 0, 0, 0 };
```

However, there was also the `ExportLevelDetails` and `ExportLevelDetailsEx` macros that were added, and which replace manually defining the DescBlock:
https://github.com/OutpostUniverse/Outpost2DLL/blob/72391d9cbe9c056a8d704ec2c2837ce36f75c6bb/RequiredExports.h#L49-L53

The equivalent code for newer levels is then something like:
```cpp
ExportLevelDetailsEx("6P, LoS, '<map name>'", "on6_01.map", "MULTITEK.TXT", MultiLastOneStanding, 6, 12, false)
```

Reply:
One additional detail. The name `AIModDesc` is part of the exported API from Outpost2.exe. However, that type name doesn't appear in any method used by levels. Rather it appears in shell related code, which was split off into the `Outpost2App.h` header file. When the `Outpost2DLL` code was first written, we hadn't yet discovered the struct being exported by the levels matched up with the `AIModDesc` struct used by the shell methods. As it was a named data export (not a C++ function export), it didn't have any type information encoded in the name. Hence it ended up getting the made up name `SDescBlock`, which I believe came from the data exports variable name `DescBlock`. The `S` was prepended for "Struct", and to make type name unique from the exported variable name.

So basically all the old levels use a duplicate made up fake name of `SDescBlock`, rather than the proper name `AIModDesc`. That was part of the reason for the upgrade.

The other part was to remove the boiler plate of exporting the `MapName`, `LevelDesc` and `TechtreeName` fields, and also setting pointers to them in the `AIModDesc`. Since it's so formulaic, why bother to burden the level writer with the details.